quinn-proto/src/connection.rs

```diff
@@ -337,6 +353,74 @@ impl Connection {
                 mux.emit(Event::StreamWritable { stream });
             }
         }
+
+        // Explicit congestion notification
+        if self.sending_ecn {
+            if let Some(ecn) = ack.ecn {
+                // We only examine ECN counters from ACKs that we are certain we received in transmit
+                // order, allowing us to compute an increase in ECN counts to compare against the number
+                // of newly acked packets that remains well-defined in the presence of arbitrary packet
+                // reordering.
+                if ack.largest > prev_largest {
+                    match self.detect_ecn(newly_acked.len() as u64, ecn) {
```

Comment:
Style: maybe extract this `match` from the deeply nested `if`-blocks prevent all the rightward drift?